wsmaster/che-core-api-factory-github/src/test/java/org/eclipse/che/api/factory/server/github/GithubURLParserTest.java

```diff
@@ -105,6 +131,21 @@ public void checkParsing(
     };
   }
 
+  @DataProvider(name = "parsingBadRepository")
+  public Object[][] parsingBadRepository() {
+    return new Object[][] {
+      {"https://github.com/eclipse/che .git", "che .git"},
+      {"https://github.com/eclipse/.git", ".git"},
+      {"https://github.com/eclipse/myB@dR&pository.git", "myB@dR&pository.git"},
+      {"https://github.com/eclipse/.", "."},
+      {"https://github.com/eclipse/івапівап.git", "івапівап.git"},
+      {"https://github.com/eclipse/ ", " "},
+      {"https://github.com/eclipse/.", "."},
+      {"https://github.com/eclipse/repositorygit", "repositorygit"},
```

Comment:
This is a perfectly valid repository name. Why is it placed in `parsingBadRepository`?

Reply:
yes it can be removed

Reply:
pls don't remove the test case, just move it to proper test method. It catches possibly very nasty bug because repository name with `git` at the end might easily happen.